packages/website/utils/content.ts

```diff
@@ -86,18 +44,57 @@ async function fetchFiles(targetPath: string) {
   return fetchedFiles;
 }
 
-async function geltAllMDX() {
-  const allMDX = await fetchFiles(componentsPath);
-  const mdxData = allMDX.map((filepath) => getSlugByFilepath(filepath));
+async function getAllMdx(paths: string[]) {
+  let allMDX: string[] = [];
 
-  // Saving a map of slugs and the filepath to their mdx in a JSON file
-  const mdxFilepathBySlug = mdxData.reduce((acc, mdx) => {
-    return { ...acc, [mdx.slug]: mdx.filepath };
-  }, {});
-  const data = JSON.stringify(mdxFilepathBySlug, null, 2);
-  fs.writeFileSync('utils/mdxFilepathBySlug.json', data);
+  for (const path of paths) {
+    const newFiles = await fetchFiles(path);
+    allMDX = allMDX.concat(newFiles);
+  }
 
+  const mdxData = allMDX
+    .map((filepath) => getDataByFilepath(filepath))
+    .filter((data) => {
+      // return only those mdx files which have 'slug' set in meta section
+      return data.frontMatter.data.slug;
+    });
   return mdxData;
 }
 
-export { geltAllMDX, getPageBySlug };
+const allMdxSources = [
+  path.resolve('../../packages/components'),
+  path.resolve('../../packages/core'),
+  path.resolve('../../packages/forma-36-website/src/content'),
+];
+
+async function getMdxSourceBySlug(slug: string[]) {
+  const mdxFiles = await getAllMdx(allMdxSources);
+  const joinedSlug = slug.join('/');
+  return mdxFiles.find(
+    (item) =>
+      item.frontMatter.data.slug === `/${joinedSlug}/` ||
+      item.frontMatter.data.slug === `/${joinedSlug}`,
+  );
```

Comment:
Why do we need this? I thought the slugs where already  a string like "/components/button"

Reply:
sometimes we have trailing slash and sometimes we don't :/